python/pysrc/kaskada/timestream/_aggregation.py

```diff
@@ -0,0 +1,215 @@
+from __future__ import annotations
+
+from typing import Optional, Union
+
+import kaskada as kd
+import pyarrow as pa
+
+from . import LiteralValue, Timestream, record
+
+
+def _aggregation(
```

Comment:
If we're going to group things like this, I wonder if we should also group and organize the tests similarly.

Reply:
I think this would be a very good idea.